### ISSUE TITLE
[Feature Request]: Use `Span<>` for all `foreach` loops

### ISSUE THREAD

**Issue by PWagner1** [Krypton-Suite/Standard-Toolkit#846]
As part of .NET 7, it is recommended to use `Span<>` instead of `foreach`, as it's generally faster.

**Comment by dax-leo** [comment]
Vote for this one! Thank you! 
Span<> should become a standard in .NET, but many are still using IEnumerable<> interface (myself included :) 
It's hard to get use to it after so many years.
Recently I updated one of my very old parsers which usually has to process billions of lines from multiple sources. After adapting code to ReadOnlySpan<> I measured over 50% perf. improvement and significantly less pressure on heap and GC activity (due to Span is always on stack).
I wanted to add this request myself then I saw yours post :) Cool ! Thnx!
